src/app/services/transaction/transaction-base/transaction-base.component.ts

```diff
@@ -0,0 +1,172 @@
+import { Component } from "@angular/core";
+import { IgxTransactionService, State, Transaction, TransactionType } from "igniteui-angular";
+import { WISHLIST, WishlistItem } from "../data";
+
+@Component({
+    selector: "transaction-base",
+    styleUrls: ["./transaction-base.component.scss"],
+    templateUrl: "transaction-base.component.html"
+})
+export class TransactionBaseComponent {
+    public wishlist: WishlistItem[];
+    public addDisabled: boolean;
+    public deleteDisabled: boolean;
+    public editDisabled: boolean;
+
+    // inect the transaction service
+    constructor(public transactions: IgxTransactionService<Transaction, State>) {
+        this.wishlist = WISHLIST;
+    }
+
+    /*
+     * For each type of action that we want to perform we can add different types of transactions.
+     * The below methods - onAdd, onEdit, onDelete demonstrate the different types of transactions
+     * and what kind of data we normally pass to each of them.
+     *
+     * Each transaction needs to have id, type and a newValue properties.
+     * ADD and DELETE transactions should have unique IDs.
+     * There can be multiple UPDATE transactions with the same ID,
+     * since there can be multiple consecutive changes on a single element.
+    */
+
+    /**
+     * Create an 'ADD' transaction.
+     */
+    public onAdd(event): void {
```

Comment:
The interesting thing about ADD transaction is that you do not provide `recordRef`, as you do not have one. Add comment about this.